src/cls/rgw/cls_rgw.cc

```diff
@@ -3219,7 +3219,8 @@ static int gc_iterate_entries(cls_method_context_t hctx, const string& marker, b
       if (max_entries && (i >= max_entries)) {
         if (truncated)
           *truncated = true;
-        key_iter = key;
+        --iter;
+        key_iter = iter->first;
```

Comment:
cls_cxx_map_get_vals(hctx, start_key, filter, GC_NUM_KEYS, &keys);    The start_key here actual means start_after. The start_key wouldn't included in keys.